app/src/main/java/com/kickstarter/libs/utils/ApplicationLifecycleUtil.java

```diff
@@ -44,24 +51,36 @@ public void onActivityStarted(final @NonNull Activity activity) {
 
   @Override
   public void onActivityResumed(final @NonNull Activity activity) {
-    if(this.isInBackground){
-
+    if(this.isInBackground) {
       // Facebook: logs 'install' and 'app activate' App Events.
       AppEventsLogger.activateApp(activity.getApplication());
 
+      //to get the current config for segment feature when app resume from background
+      this.config.observable().subscribe(c -> {
+        this.currentConfig = c;
+      });
+
       // Refresh the config file
       this.client.config()
-        .compose(Transformers.pipeApiErrorsTo(this::handleConfigApiError))
-        .compose(Transformers.neverError())
-        .subscribe(c -> this.config.config(c));
+              .compose(Transformers.pipeApiErrorsTo(this::handleConfigApiError))
+              .compose(Transformers.neverError())
+              .subscribe(c -> {
+                if (this.build.isDebug() || Build.isInternal()) {
+                  if (Objects.requireNonNull(c.features()).containsKey(ConfigFeatureName.SEGMENT_ENABLED.getConfigFeatureName())) {
+                    final boolean isChecked = Objects.requireNonNull(this.currentConfig.features()).get(ConfigFeatureName.SEGMENT_ENABLED.getConfigFeatureName());
```

Comment:
I'm gonna suggest a few things in here, but I'll be happy to Pair program with you if it's not clear :) 
1 - when a feature flag is checked storage a user preference with the name of the feature flag and the checked/uncheck value
2 - remove the lines 58 to 61
3 - for overriding the value instead of checking the previous config file in memory just check the user preference for that feature flag value

You can even create a Config extension method that will check for each feature flag if there is an associated user preference value.

Reply:
Done